src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -1561,6 +1561,29 @@ def test_list_compute_skus_partially_unavailable(self):
         result = self.cmd('vm list-skus -l eastus --query "[?name==\'Standard_M64m\']"').get_output_in_json()
         self.assertTrue(result and result[0]["restrictions"] and result[0]["restrictions"][0]["reasonCode"] == 'NotAvailableForSubscription')
 
+    @AllowLargeResponse(size_kb=99999)
+    def test_list_compute_skus_partially_unavailable2(self):
+        results = self.cmd('vm list-skus').get_output_in_json()
+        for result in results:
+            if 'restrictions' not in result:
+                continue
+            is_available = True
+            is_restrict_zone = False
+            is_restrict_location = False
+            for restriction in result['restrictions']:
+                if restriction['reasonCode'] != 'NotAvailableForSubscription':
+                    break
+                if restriction['type'] == 'Zone' \
+                        and not (set(result['locationInfo'][0]['zones'] or []) -
+                                 set(restriction['restrictionInfo']['zones'] or [])):
+                    is_restrict_zone = True
+                if restriction['type'] == 'Location' and (
+                        result['locationInfo'][0]['location'] in (restriction['restrictionInfo']['locations'] or [])):
+                    is_restrict_location = True
+                if is_restrict_zone and is_restrict_location:
```

Comment:
I think it is possible for a VM sku to be restricted in region but not to have zonal restrictions listed. So I think the SKU should be considered restricted if it has a "Location" restriction OR all zones restricted.

Reply:
> I think it is possible for a VM sku to be restricted in region but not to have zonal restrictions listed. So I think the SKU should be considered restricted if it has a "Location" restriction OR all zones restricted.

Thanks. We modified the code again according to your comment. Could you please help to confirm it again?

Reply:
I thought about this a bit more and think we may want to bring in Azure Compute team into conversation here to double-check.

If we are filtering out VM sizes from the list based on the restriction types, we may need to make the filtering logic also dependent on the --zone parameter which for this CLI command is described as "--zone: Show skus supporting availability zones.  Allowed values: false, true"

I put together the following table to see if it makes sense to you and to Azure Compute:

| Restriction Types | VM SKU Availability | is_available?  |
|---|---|---|
| Location=null, Zone=null | Yes, regional or any zone | TRUE  |
| Location=null, Zone=1,2 | Yes, regional or zone 3 | TRUE  |
| Location=null, Zone=1,2,3 | Yes, but only regional | if (--zone parameter) then FALSE else TRUE  |
| Location=eastus2, Zone=null | No, since region is restricted all zones are restricted | FALSE  |
| Location=eastus2, Zone=1,2 | No, since region is restricted all zones are restricted | FALSE  |
| Location=eastus2, Zone=1,2,3 | No, since region is restricted all zones are restricted | FALSE  |

Reply:
> I thought about this a bit more and think we may want to bring in Azure Compute team into conversation here to double-check.
> 
> If we are filtering out VM sizes from the list based on the restriction types, we may need to make the filtering logic also dependent on the --zone parameter which for this CLI command is described as "--zone: Show skus supporting availability zones. Allowed values: false, true"
> 
> I put together the following table to see if it makes sense to you and to Azure Compute:
> 
> Restriction Types	VM SKU Availability	is_available?
> Location=null, Zone=null	Yes, regional or any zone	TRUE
> Location=null, Zone=1,2	Yes, regional or zone 3	TRUE
> Location=null, Zone=1,2,3	Yes, but only regional	if (--zone parameter) then FALSE else TRUE
> Location=eastus2, Zone=null	No, since region is restricted all zones are restricted	FALSE
> Location=eastus2, Zone=1,2	No, since region is restricted all zones are restricted	FALSE
> Location=eastus2, Zone=1,2,3	No, since region is restricted all zones are restricted	FALSE

Hello @grizzlytheodore, could you please help to confirm this logic about filtering out VM sizes from the list based on the restriction types? Thanks.

Hello @chasewilson, if the relationship mentioned above is confirmed, we would modify the logic about filtering out VM sizes from the list based on the restriction types.

Reply:
Hello @grizzlytheodore, could you please help to confirm this logic?

> I thought about this a bit more and think we may want to bring in Azure Compute team into conversation here to double-check.
> 
> If we are filtering out VM sizes from the list based on the restriction types, we may need to make the filtering logic also dependent on the --zone parameter which for this CLI command is described as "--zone: Show skus supporting availability zones. Allowed values: false, true"
> 
> I put together the following table to see if it makes sense to you and to Azure Compute:
> 
> Restriction Types	VM SKU Availability	is_available?
> Location=null, Zone=null	Yes, regional or any zone	TRUE
> Location=null, Zone=1,2	Yes, regional or zone 3	TRUE
> Location=null, Zone=1,2,3	Yes, but only regional	if (--zone parameter) then FALSE else TRUE
> Location=eastus2, Zone=null	No, since region is restricted all zones are restricted	FALSE
> Location=eastus2, Zone=1,2	No, since region is restricted all zones are restricted	FALSE
> Location=eastus2, Zone=1,2,3	No, since region is restricted all zones are restricted	FALSE

@travismaier, could you please help to confirm this logic about filtering out VM sizes from the list based on the restriction types? Thanks.